content/doc/book/installing/_docker.adoc

```diff
@@ -189,7 +188,7 @@ copying and pasting this annotation-free version here:
 +
 [source,bash,subs="attributes+"]
 ----
-docker run --name jenkins-blueocean --rm --detach \
+docker run --name jenkins-blueocean --restart=always --detach \
```

Comment:
`always` or `on-failure`? (need to check if you do shutdown from the UI whether it exits or just restarts with always)

Reply:
When I ask for shutdown within the UI `Manage Jenkins` `->` `Prepare for Shutdown` it never shutdowns with Docker, with `always` or `on-failure`, or even with `--rm`.
If I use the `/safeRestart` URI though, the docker container stops and never comes back with `--rm`, but restarts with `always` or `on-failure`.

Reply:
any thoughts @basil?

Reply:
The `/exit` endpoint works fine for me too, it's just the workflow `Manage Jenkins -> Prepare for Shutdown` that does not work for me.

Reply:
If you think you have found a bug, [this page](https://www.jenkins.io/participate/report-issue/) discusses what to do.

Reply:
I'm seeing the same behavior that @basil describes.  I prefer the `--restart=on-failure` behavior.

When I used `--restart=on-failure` with docker running in the foreground from my terminal, the `/restart` end point stopped the docker process and then it was restarted in the background.  I could see it was still running with `docker ps` and I could connect to it from my web browser.

When I used `--restart=on-failure` with docker running in the foreground from my terminal, the `/exit` end point stopped the docker process and it was not restarted in the background.  I think that is the behavior we want.

When I used the `--restart=always` with docker running in the foreground from my terminal, the `/restart` stopped the docker process and then it was restarted in the background.

When I used the `--restart=always` with docker running in the foreground from my terminal, the `/exit` stopped the docker process and then it was restarted in the background.  I don't think we want that behavior.

Reply:
Thanks @MarkEWaite . I have switched to `on-failure` 👍